Makefile

```diff
@@ -78,12 +99,22 @@ depend.status: Gopkg.lock ; $(info $(H) reporting dependencies status...)
 	$(Q) dep status
 
 # @todo only run if there are changes (e.g., create a checksum file?) 
+# Update the vendor dir, pulling latest compatible dependencies from the
+# defined branches.
 depend.ensure: Gopkg.lock ; $(info $(H) ensuring dependencies are up to date...)
 	$(Q) dep ensure
 
 depend.graph: Gopkg.lock ; $(info $(H) visualizing dependency graph...)
 	$(Q) dep status -dot | dot -T png | display
 
+# Re-create the vendor directory, if it doesn't exist, using the checked in lock file
+depend.vendor: vendor
+	$(Q) dep ensure -vendor-only
+
+vendor:
```

Comment:
Does this need a .PHONY too?

Reply:
Vendor is not phony I think - if the dir is missing we need to run dep ensure. 

But likely there are missing tags in the .phony, will go over the file again after I get all
the tests passing ( and probably in separate PRs)